### PR TITLE
Make isFluidHandle public

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

This makes isFluidHandle public.

An alternative API, using Symbol.hasInstrance and `instancof IFluidHandle` is not possible here, since there already is non-type value exported under the name IFluidHandle (its the string "IFluidHandle") which can not have the symbol added to it.

This leaves the existing isFluidHandle function seeming like the only real API option.

Detecting handles correctly requires knowing implementation details of handle and dealing with the possibility of old in memory handle representations, so we don't want customers rolling their own. As there are use-cases for handle detection, we should probably expose this API to do it robustly.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

Do we want to make this public? I think so, but that should be covered by the review. We could also go to alpha or beta first. THis seems pretty safe to stabilize though.

### CODE REVIEW

File: packages/framework/fluid-framework/src/index.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ export type {
 	FluidObjectProviderKeys, // Used by FluidObject
 } from "@fluidframework/core-interfaces";
 
+export type { isFluidHandle } from "@fluidframework/runtime-utils";
```
Comment (tylerbutler) [inline packages/framework/fluid-framework/src/index.ts:59]: This is a new export source for fluid-framework. @Josmithr do we need to make other config changes to ensure the docs are built correctly?

Comment (Josmithr) [reply packages/framework/fluid-framework/src/index.ts:59]: Nope :) API-Extractor in this package is configured to bundle all `@fluidframework` packages automatically (assuming they are correctly declared as a direct dependency, which this now is).

Comment (Josmithr) [reply packages/framework/fluid-framework/src/index.ts:59]: In this case you can tell because the actual declaration ends up in the API reports in this package, rather than appearing as a re-export from an external package.

File: packages/runtime/runtime-utils/src/handles.ts [hunk 2]
```diff
@@ -39,10 +39,10 @@ export const isSerializedHandle = (value: any): value is ISerializedHandle =>
 const enableBackwardsCompatibility = true;
 
 /**
- * Check if a value is an IFluidHandle.
+ * Check if a value is an `IFluidHandle`.
```
Comment (Josmithr) [inline packages/runtime/runtime-utils/src/handles.ts:42]: Nit
```suggestion
 * Check if a value is an {@link @fluidframework/core-interfaces#IFluidHandle}.
```

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -1818,130 +1818,6 @@ importers:
         specifier: ^5.8.0
         version: 5.10.0
 
-  examples/benchmarks/bubblebench/shared-tree-flex-tree:
```
Comment (Josmithr) [inline pnpm-lock.yaml:1821]: Nit: These changes seem unrelated.

**Review by Josmithr — APPROVED** [review]

File: pnpm-lock.yaml (hunk 3 above)
Comment (noencke) [reply pnpm-lock.yaml:1821]: It looks like they are from a PR I did the other day, where I removed the shared-tree-flex-tree package. So this is probably good to check in. Is there something I need to do besides run `pnpm i` in order to regenerate the pnpm-lock? I thought that running `pnpm i` would have done this for me, but I'm pretty sure I ran `pnpm i` and it did not.

Comment (CraigMacomber) [reply pnpm-lock.yaml:1821]: If you want the lock file to update, you need to run `pnpm i --no-frozen-lockfile` now. That said if doing so would make a change, I'd expect pnpm i to error and tell you that,. I guess maybe there is some subset of changes which would update, but doesn't error?
